src/rust/ide/src/controller/searcher.rs

```diff
@@ -322,49 +343,49 @@ pub struct Searcher {
     database        : Rc<model::SuggestionDatabase>,
     language_server : Rc<language_server::Connection>,
     parser          : Parser,
-    this            : Rc<Option<ThisNode>>,
+    this_arg        : Rc<Option<ThisNode>>,
 }
 
 impl Searcher {
     /// Create new Searcher Controller.
     pub async fn new
-    ( parent   : impl AnyLogger
-    , project  : &model::Project
-    , method   : language_server::MethodPointer
-    , this     : Option<double_representation::node::Id>
+    ( parent         : impl AnyLogger
+    , project        : &model::Project
+    , method         : language_server::MethodPointer
+    , mode           : Mode
+    , selected_nodes : Vec<double_representation::node::Id>
     ) -> FallibleResult<Self> {
         let graph = controller::ExecutedGraph::new(&parent,project.clone_ref(),method).await?;
-        Ok(Self::new_from_graph_controller(parent,project,graph,this))
+        Self::new_from_graph_controller(parent,project,graph,mode,selected_nodes)
     }
 
-    // /// Check if the picked fragment is still unmodified by user.
-    // pub fn is_fragment_unmodified(&self, fragment:&FragmentAddedByPickingSuggestion) -> bool {
-    //     let this_var      = self.this_var_for(fragment.id);
-    //     let default_code  = fragment.picked_suggestion.code_to_insert(this_var);
-    //     let fragment_code = self.data.borrow().input.completed_fragment(fragment.id);
-    //     fragment_code.contains(&default_code.as_str())
-    // }
-
     /// Create new Searcher Controller, when you have Executed Graph Controller handy.
     pub fn new_from_graph_controller
-    ( parent  : impl AnyLogger
-    , project : &model::Project
-    , graph   : controller::ExecutedGraph
-    , this    : Option<double_representation::node::Id>
-    ) -> Self {
-        let logger = Logger::sub(parent,"Searcher Controller");
-        let this   = Rc::new(this.and_then(|id| ThisNode::new(id,&graph.graph())));
-        let this   = Self {
-            logger,graph,this,
-            data            : default(),
+    ( parent         : impl AnyLogger
+    , project        : &model::Project
+    , graph          : controller::ExecutedGraph
+    , mode           : Mode
+    , selected_nodes : Vec<double_representation::node::Id>
+    ) -> FallibleResult<Self> {
+        let logger   = Logger::sub(parent,"Searcher Controller");
+        let database = project.suggestion_db();
+        let data     = if let Mode::EditNode{node_id} = mode {
+            Data::new_with_edited_node(&graph.graph(),&*database,node_id)?
+        } else {
+            default()
+        };
+        let this_arg = Rc::new(ThisNode::new(selected_nodes,&graph.graph()));
```

Comment:
this_arg could be initialized directly in the structure.

Reply:
Tried but doesn't work well here, as in the structure the `graph` is moved. So we'd end up mixing both syntaxes for field initialization that doesn't look good.
Leaving as-is.